src/uu/tr/src/operation.rs

```diff
@@ -0,0 +1,529 @@
+            },
+        )
+    }
+
+    fn parse_alnum(input: &str) -> IResult<&str, Sequence> {
+        tag("[:alnum:]")(input).map(|(l, _)| {
+            (
+                l,
+                Sequence::CharRange(('a'..='z').chain('A'..'Z').chain('0'..'9').collect()),
+            )
+        })
+    }
+
+    fn parse_alpha(input: &str) -> IResult<&str, Sequence> {
+        tag("[:alpha:]")(input).map(|(l, _)| {
+            (
+                l,
+                Sequence::CharRange(('a'..='z').chain('A'..'Z').collect()),
+            )
+        })
+    }
+
+    fn parse_blank(input: &str) -> IResult<&str, Sequence> {
+        tag("[:blank:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_control(input: &str) -> IResult<&str, Sequence> {
+        tag("[:cntrl:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_digit(input: &str) -> IResult<&str, Sequence> {
+        tag("[:digit:]")(input).map(|(l, _)| (l, Sequence::CharRange(('0'..='9').collect())))
+    }
+
+    fn parse_graph(input: &str) -> IResult<&str, Sequence> {
+        tag("[:graph:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_lower(input: &str) -> IResult<&str, Sequence> {
+        tag("[:lower:]")(input).map(|(l, _)| (l, Sequence::CharRange(('a'..='z').collect())))
+    }
+
+    fn parse_print(input: &str) -> IResult<&str, Sequence> {
+        tag("[:print:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_punct(input: &str) -> IResult<&str, Sequence> {
+        tag("[:punct:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_space(input: &str) -> IResult<&str, Sequence> {
+        tag("[:space:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_upper(input: &str) -> IResult<&str, Sequence> {
+        tag("[:upper:]")(input).map(|(l, _)| (l, Sequence::CharRange(('A'..='Z').collect())))
+    }
+
+    fn parse_xdigit(input: &str) -> IResult<&str, Sequence> {
+        tag("[:xdigit:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_char_equal(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("[="), take(1usize), tag("=]")))(input).map(|(_, (_, _, _))| todo!())
+    }
+}
+
+pub trait SymbolTranslatorNew {
+    fn translate(&mut self, current: char) -> Option<char>;
+}
+
+#[derive(Debug, Clone)]
+pub struct DeleteOperationNew {
+    set: Vec<Sequence>,
+    complement_flag: bool,
+}
+
+impl DeleteOperationNew {
+    pub fn new(set: Vec<Sequence>, complement_flag: bool) -> DeleteOperationNew {
+        DeleteOperationNew {
+            set,
+            complement_flag,
+        }
+    }
+}
+
+impl SymbolTranslatorNew for DeleteOperationNew {
+    fn translate(&mut self, current: char) -> Option<char> {
+        let found = self.set.iter().any(|sequence| match sequence {
+            Sequence::Char(c) => c.eq(&current),
+            Sequence::CharRange(r) => r.iter().any(|c| c.eq(&current)),
+        });
+        (self.complement_flag == found).then(|| current)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum TranslateOperationNew {
```

Comment:
Since you removed all the old operations, I think it's safe to remove the `"New"` everywhere

Reply:
Yep.